internal/test_cases/type_of_command_test_case.go

```diff
@@ -0,0 +1,76 @@
+package test_cases
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+type TypeOfCommandTestCase struct {
+	Command        string
+	SuccessMessage string
+}
+
+func (t *TypeOfCommandTestCase) RunForBuiltin(asserter *logged_shell_asserter.LoggedShellAsserter, shell *shell_executable.ShellExecutable, logger *logger.Logger) error {
+	if t.SuccessMessage == "" {
+		t.SuccessMessage = "✓ Received expected response"
+	}
+
+	testCase := CommandResponseTestCase{
+		Command:          fmt.Sprintf("type %s", t.Command),
+		ExpectedOutput:   fmt.Sprintf(`%s is a shell builtin`, t.Command),
+		FallbackPatterns: []*regexp.Regexp{regexp.MustCompile(fmt.Sprintf(`^%s is a( special)? shell builtin$`, t.Command))},
+		SuccessMessage:   t.SuccessMessage,
+	}
+
+	if err := testCase.Run(asserter, shell, logger); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *TypeOfCommandTestCase) RunForExecutable(asserter *logged_shell_asserter.LoggedShellAsserter, shell *shell_executable.ShellExecutable, logger *logger.Logger, customExecutablePath string) error {
+	var expectedPath string
+
+	if t.Command == "my_exe" {
```

Comment:
@andy1li this seems like a hardcoded case, can we avoid it?

Reply:
In the previous version this made sense since `my_exe` was created in the same function and that context was present. Here it doesn't - how is the caller supposed to know that the function behaves differently based on the special string "my_exe"?

Reply:
My bad. Let me refactor it.

Reply:
Refactored.